recipes/zstd/all/test_package/conanfile.py

```diff
@@ -25,3 +28,16 @@ def test(self):
             bin_path = os.path.join(self.cpp.build.bindirs[0], "test_package")
             png_file = os.path.join(self.source_folder, "logo.png")
             self.run(f"{bin_path} {png_file}", env="conanrun")
+
+            zstd_dep = self.dependencies["zstd"]
+            if(zstd_dep.options.build_programs):
+                zstd = os.path.join(zstd_dep.package_folder, zstd_dep.cpp_info.bindirs[0], "zstd")
+                zstd_version = zstd_dep.ref.version
+
+                output = StringIO()
+                self.run(f"{zstd} --version", output, env="conanrun")
+                s = re.search(f"{zstd_version}", output.getvalue())
```

Comment:
```suggestion
                zstd_version = zstd_dep.ref.version

                output = StringIO()
                self.run("zstd --version", output)
                s = re.search(zstd_version, output.getvalue())
```
The `zstd` executable should already be available on `PATH` thanks to the `VirtualRunEnv` generator.

Reply:
That was my own initial thought. It will however not work if you build the package on a system that has zstd installed, because then that one will be executed.

`self.run(f"which zstd")` produces:

```
zstd/1.5.5 (test package): RUN: which zstd
/usr/bin/zstd
```

with or without `env="conanrun"`

Reply:
Oh, yes, that's a very valid point.

Reply:
Oh, yes, that's a very valid point.